tools/bin/ariaDownload.py

```diff
@@ -260,45 +256,60 @@ def make_nc_cookie(self):
 
-    if nt == 'all':
+    if inps.num_threads == 'all':
         nt = max_threads
     else:
-        nt = int(nt)
-
-    nt = nt if nt < max_threads else max_threads
+        nt = int(inps.num_threads)
+    nt     = nt if nt < max_threads else max_threads
     log.info('Using  %s threads for parallel downloads', nt)
-    chunks = np.array_split(downloader.files, nt)
-    chunks = [chunk for chunk in chunks if chunk.size > 0]
-    chunk_check = []
-    # sanity check
-    for chunk in chunks: chunk_check.extend(chunk)
-    if not chunk_check == downloader.files:
-        log.error('Numpy incorrectly split product list; run serially')
-        raise RuntimeError
-
-    # the ASF downloader obj must be called again in here otherwise multiprocessing fails
-    with multiprocessing.Pool(nt) as pool:
-         pool.map(_dl_helper, chunks)
-
-def _dl_helper(files):
+
+    files          = dct_prod['product_list'].split(',')
+    chunks1        = np.array_split(files, np.ceil(len(files)/200)) # split by 200s
+    check          = []
+    for chunk in chunks1:
+        chunks     = np.array_split(chunk, nt)
+        lst_dcts   = [vars(inps).copy() for i in range(nt)]  # Namespace->dictionary, repeat it
+        for i, chunk1 in enumerate(chunks):
+            lst_dcts[i]['files'] = chunk1 # put split up files to the objects for threads
+            lst_dcts[i]['ext']   = i      # for unique bulk downloader file name
+
+        with multiprocessing.Pool(nt) as pool:
+            try:
+                info = pool.map(_dl_helper, lst_dcts)
+            except Exception as E:
+                print ('ASF bulk downloader error:', E)
+                print ('Likely a bad handshake with the ASF DAAC. Try rerunning')
+                os.sys.exit(1)
+
+        check.extend(chunkc for chunkc in chunks) # in case products missed in split
+
+    rewrite_summary(info)
+
+    check = [item for sublist in check for item in sublist]
+    for f in files:
+        if not f in check:
+            log.critical('File splitting missed: %s; subset your ARIA in command in time to try again', f)
+
+def _dl_helper(inp_dct):
     """ Helper function for parallel processing """
-    import ASFDataDload as AD
-    os.sys.argv = [] # gets around spurious messages
-    downloader = AD.bulk_downloader()
-    files = [files] if isinstance(files, str) else files
-    downloader.files = files
-    downloader.download_files()
-    downloader.print_summary()
+    prod_dct = {'product_list': ','.join(inp_dct['files'])}
+    log.debug ('# of files: %d', len(inp_dct['files']))
+    script   = requests.post(f'{inp_dct["url_base"]}&output=Download', data=prod_dct).text
+
+    fname    = f'ASFDataDload{inp_dct["ext"]}.py'
+    fpath    = os.path.abspath(os.path.join(inp_dct['wd'], fname))
+    with open(fpath, 'w') as f: f.write(script)
+    AD = __import__(os.path.splitext(fname)[0])
+    args, os.sys.argv = os.sys.argv, [] # gets around spurious messages
```

Comment:
Codacy found an issue: [local variable 'args' is assigned to but never used (F841)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6310391)